lib/record-events-browser.js

```diff
@@ -169,239 +122,181 @@ export function recordEvents(eventsHash, callback){
   this.emit('recordEvents', extendedEventsHash);
 
   if (!Keen.enabled) {
-    handleValidationError.call(this, 'Keen.enabled is set to false.', cb);
+    handleValidationError.call(this, 'Keen.enabled is set to false.', callback);
     return false;
   }
 
-  if (getXhr()) {
-    sendXhr.call(this, 'POST', url, extendedEventsHash, cb);
+  return send.call(this, { url, extendedEventsHash, callback });
+}
+
+function send({ url, extendedEventsHash, callback, configObject = {}, eventCollection}){
+  const clientConfig = this.config;
+  const requestType = configObject.requestType // specific method for one request
+    || this.config.requestType; // global request type of client
+
+  if (
```

Comment:
@danwiding does this section covered what you were trying to do here? https://github.com/ProdPerfect/prodperfect-keen-tracking.js/commit/858340e91744244abca12112e0bcdbe10a9373b2#diff-86b048434ef97f9f2989abb3694de37bR80

Reply:
I'm not certain this works. Though I can't tell. I need to be walked through this synchronously.